### ISSUE TITLE
Uninitialized value in tftpboot-check.pl

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#429]
**Reported by bli on 27 Sep 2007 01:58 UTC**
When installing OSCAR on an unsupported distribution gives:

```
# ./install_cluster --cli eth0
--> Setting env variable OSCAR_UI to cli
--> Setting env variable OSCAR_HOME to /root/download.d/oscar-5.1a1r6218nightly-20070925
Checking your system...
Success: debrepo-check.pl
Success: display-check.pl
 --------------------------------------------------
 ERROR: it seems the interface used is not the one
 assigned to OSCAR in /etc/hosts
 --------------------------------------------------
 -----------------------------------
  /root/download.d/oscar-5.1a1r6218nightly-20070925/scripts/system-sanity.d/network-check.pl
  Network configuration not correct
 -----------------------------------
Failure: network-check.pl
Success: selinux-check.pl
Success: ssh-check.pl
Success: su-check.pl
Use of uninitialized value in concatenation (.) or string at /root/download.d/oscar-5.1a1r6218nightly-20070925/scripts/system-sanity.d/tftpboot-check.pl line 19.
Use of uninitialized value in concatenation (.) or string at /root/download.d/oscar-5.1a1r6218nightly-20070925/scripts/system-sanity.d/tftpboot-check.pl line 19.
Use of uninitialized value in concatenation (.) or string at /root/download.d/oscar-5.1a1r6218nightly-20070925/scripts/system-sanity.d/tftpboot-check.pl line 19.
 ---------------------------------------------------------
 ERROR: Impossible to find a local or online repository in
 /tftpboot for the distro --
 ---------------------------------------------------------
Failure: tftpboot-check.pl
ERROR: There are basic system configuration issues (see log).
```

`tftpboot-check.pl` failed to open `OS_Detect` and cause the above problems.

This is CentOS 5.0 x86 BTW.

See also RFE #428.

**Comment by dikim33** [comment]
**Comment by valleegr on 8 Nov 2007 02:29 UTC**
Why is this a bug if the distro is not supported? Do you mean we should have a separate test to check if the distro is supported?

**Comment by dikim33** [comment]
**Comment by valleegr on 6 Feb 2008 18:44 UTC**
(In [6921]) This check-in fixes #429 - If the OS is not correctly detected we cleanly exist.
